### PR TITLE
Refactor exporter config

### PR DESCRIPTION
**Pull request by SuperQ**
Switch exporter over to standard Prometheus exporter flags and logging. This eliminates the need for a configuraion file.

Closes: https://github.com/prometheus-community/smartctl_exporter/issues/66
Closes: #30 

Signed-off-by: SuperQ <superq@gmail.com>

### CODE REVIEW

**Comment by k0ste** [comment]
Resolves #30

**Comment by k0ste** [comment]
Seems this also doesn't support glob in device param
This is useful, when i want to exclude `scsi` and other devices, and use only `nvme` devices (and don't know how much devices will be present on host)

```golang
[root@host tmp]# ./smartctl_exporter --smartctl.interval=1h --log.level=debug --smartctl.device=/dev/nvme*
ts=2022-09-30T11:20:26.215Z caller=main.go:94 level=info msg="Starting systemd_exporter" version="(version=, branch=, revision=)"
ts=2022-09-30T11:20:26.215Z caller=main.go:95 level=info msg="Build context" build_context="(go=go1.17.2, user=, date=)"
ts=2022-09-30T11:20:26.215Z caller=main.go:57 level=error msg="Error collecting SMART data" err="Device /dev/nvme* unavialable"
panic: runtime error: index out of range [0] with length 0

goroutine 30 [running]:
main.(*SMARTctlInfo).mineVersion(0xc000185e08)
        /smartctl_exporter/smartctlinfo.go:59 +0xbaf
main.(*SMARTctlInfo).Collect(...)
        /smartctl_exporter/smartctlinfo.go:48
main.SMARTctlManagerCollector.Collect({{0x0, 0x0}, 0x0, {0xc00021c1d0, 0x1, 0x1}, {0xa0c9a0, 0xc000135b40}}, 0xc00010e1e0)
        /smartctl_exporter/main.go:60 +0xd2
github.com/prometheus/client_golang/prometheus.DescribeByCollect.func1()
        /go/pkg/mod/github.com/prometheus/client_golang@v1.13.0/prometheus/collector.go:90 +0x2b
created by github.com/prometheus/client_golang/prometheus.DescribeByCollect
        /go/pkg/mod/github.com/prometheus/client_golang@v1.13.0/prometheus/collector.go:89 +0xa5
```

**Comment by SuperQ** [comment]
We could do globs, but I would prefer to add that in a a separate feature change.

**Comment by k0ste** [comment]
> We could do globs, but I would prefer to add that in a a separate feature change.

Created issue #72 for this

File: go.mod [hunk 1]
```diff
@@ -3,21 +3,35 @@ module github.com/prometheus-community/smartctl_exporter
 go 1.17
```
Comment (Nexucis) [inline go.mod:3]: go 1.18 ? Or do you prefer to upgrade in a different PR ?

**Review by Nexucis — APPROVED** [review]
That's definitely a nice refactoring.
I'm just wondering how much the breaking change in term of flags and config can be an issue.
By looking at the version, we are allowed to break everything but if it's widely used, perhaps some people will complain.
I'm approving this PR because I really think it worths to merge it comparing to the breaking changes it created, which just change the way to use not the way it works.

File: go.mod (hunk 1 above)
Comment (SuperQ) [reply go.mod:3]: That is just the minimum Go required. The actual Go version is in the CI/promu config.

Comment (Nexucis) [reply go.mod:3]: yeah I know, I'm just following what is done in Prometheus. Upgrading minimum go required with latest - 1 go version

**Comment by SuperQ** [comment]
Yes, there's always going to be some complaints, but there's also requests the other way, which is "Can we have this work with flags rather than a config file.

Flags are the standard across our ecosystem, and part of donating exporters to Prometheus Community is to bring them up to our standards.

**Comment by Nexucis** [comment]
sounds good to me. I forgot the project has been donated to prometheus. That's a good excuse to break thing hehe
